File: node/generate-pdf/src/faker.js
import { faker } from '@faker-js/faker';

export function generateFakeOrder() {
  const items = Array.from(
    { length: faker.number.int({ min: 1, max: 5 }) },
    () => ({
      description: faker.commerce.productName(),
      quantity: faker.number.int({ min: 1, max: 10 }),
      cost: faker.commerce.price(),
    })
  );

  return {
    id: faker.string.uuid(),
    date: faker.date.past(),
    name: faker.person.fullName(),
    items,
    total: items.reduce((acc, { cost }) => acc + parseFloat(cost), 0),
  };
}

File: python/generate-pdf/src/faker.py
from faker import Faker
import random

fake = Faker()

def generate_fake_order():
    items_count = random.randint(1, 5)
    items = [
        {
            "description": fake.product_name(),
            "quantity": random.randint(1, 10),
            "cost": float(fake.price())
        } for _ in range(items_count)
    ]
    
    total = sum(item['cost'] for item in items)
    
    return {
        "id": fake.uuid4(),
        "date": fake.past_date(),
        "name": fake.name(),
        "items": items,
        "total": total
    }
